test/e2e/scheduling/preemption.go

```diff
@@ -125,4 +125,125 @@ var _ = SIGDescribe("SchedulerPreemption [Serial] [Feature:PodPreemption]", func
 			Expect(livePod.DeletionTimestamp).To(BeNil())
 		}
 	})
+
+	// This test verifies that when a high priority pod is pending and its
+	// scheduling violates a medium priority pod anti-affinity, the medium priority
+	// pod is preempted to allow the higher priority pod schedule.
+	// It also verifies that existing low priority pods are not preempted as their
+	// preemption wouldn't help.
+	It("validates pod anti-affinity works in preemption", func() {
```

Comment:
Isn't this a pretty expensive test if you have a large cluster, do we want it defaulted on or behind [Feature] ?

Reply:
It is behind [Feature] right now. How can we prevent it from being used in scale tests?

Reply:
Ahh I scrolled back and see it now... 

Is there any reason you need to hit * nodes vs a subset?  It's still really expensive.

Reply:
Yes, there is a reason. In order to trigger preemption, cluster should be under resource pressure.

Reply:
Can't you use node selectors to force it?

Reply:
Yes, I could, but the current scenario is better, as it also exercises node scoring and ensures that we preempt the right pod no matter where in the cluster it is scheduled and also makes sure that other pods are not impacted. Node selector would remove these benefits.
What I heard is that the CI clusters typically have 4-5 nodes, except those that run scalability tests. If this is true, then the test shouldn't be very time consuming.

Reply:
What I'm really concerned about is the e2e tests are used in the field to validate clusters.  There are times when they want to validate that features are working properly.  If I was company (X) trying to validate a 200 node cluster you'd be loading their cluster and this is a very expensive test.  The usage of the e2e's goes well beyond upstream CI.

Reply:
Perhaps using a [Disruptive] tag on this test.

Reply:
I see your point. I just changed the test to create a maximum of 4 pods.

Reply:
thx for the change.